### PR TITLE
[sr] Mobile bento support

### PR DESCRIPTION
**Pull request by rgclayton**
* Adds opt in mobile stacking feature (`stack-mobile`). Mobile only feature.
* Removed border per design ask - originally added in [this PR](https://github.com/adobecom/milo/pull/6228)
* Bento aspect ratio update to 1 / 1.1
* Supports sticky title when authored with a rich-content block as the first element in a bento section.

Resolves: [MWPW-199414](https://jira.corp.adobe.com/browse/MWPW-199414)

**Test URLs:**
- Before: https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/fragments/cpro-hub?milolibs=site-redesign-foundation&mep=%2Fcc-shared%2Ffragments%2Ftests%2F2026%2Fq3%2Face1209%2Face1209-cpro-redesign.json--all
- After: https://main--da-cc--adobecom.aem.page/cc-shared/fragments/tests/2026/q3/ace1209/fragments/cpro-hub?milolibs=sr-mobile-bento&mep=%2Fcc-shared%2Ffragments%2Ftests%2F2026%2Fq3%2Face1209%2Face1209-cpro-redesign.json--all

image

### CODE REVIEW

File: libs/mep/ace1209/rich-content/rich-content.css [hunk 1]
```diff
@@ -381,8 +361,13 @@
 }
 
 @media (width >= 1440px) {
+  .rich-content.indent {
+    margin-block-end: 0;
+  }
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/rich-content/rich-content.css]: Isn't this already inherited from https://github.com/adobecom/milo/pull/6322/changes#diff-547cb21a4c0facdd7c96a9b877aba9c3178e494425359797b88fda171f80a40fR292-R294 ?

File: libs/mep/ace1209/rich-content/rich-content.css [hunk 2]
```diff
@@ -562,6 +547,44 @@
...
 
+@keyframes stack-title-stagger {
+  0% {
+    opacity: 0;
+    transform: translate3d(0, var(--parallax-translate-y), 0);
+  }
+
+  100% {
+    opacity: 1;
+    transform: translate(0, 0);
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/rich-content/rich-content.css:554]: I see a mix between `translate3d` and translate, maybe we can stick to one or the other?

File: libs/mep/ace1209/section-metadata/section-metadata.css [hunk 3]
```diff
@@ -2,6 +2,115 @@
...
+  100% {
+    transform: translateY(calc(var(--stack-offset) - var(--offset-adjuster)));
+    opacity: 1;
+  }
+}
+
+@keyframes stack-section-exit {
+  0% { transform: translate(0, 0); }
+  100% { transform: translate3d(0, calc(-1 * var(--section-exit-y)), 0); }
+}
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.css:53]: Let's stick to just `translate` for consistency, as nowadays this alone pushes things to the compositor layer.

File: libs/mep/ace1209/section-metadata/section-metadata.css [hunk 4]
```diff
@@ -2,6 +2,115 @@
...
+.section.bento.stack-mobile {
+  --phone-h: 775px;
+  --phone-vh: calc(var(--phone-h) / 100);
+  --gnav-offset: 72px;
+
+  /* Single knob for design fine-tuning — everything pins at --stack-pin-top. */
+  --bento-stack-top-gap: 40px;
+  --stack-pin-top: calc(var(--gnav-offset) + var(--bento-stack-top-gap));
+  --title-height: 24px;
+  --title-card-gap: var(--s2a-spacing-24, 24px);
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.css]: We know the token exists, so the fallback likely doesn't bring anything useful.

File: libs/mep/ace1209/section-metadata/section-metadata.css [hunk 5]
```diff
@@ -2,6 +2,115 @@
...
+
+  /* To re-enable, set this back to 40px; the stack-section-exit animation
+     + --section-vt timeline are kept intact */
+  --section-exit-y: 0;
+  --paralax-easing: cubic-bezier(0.42, 0, 0, 1);
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.css]: This is already defined in our root. However, there's a small typo here - it should be `parallax`, double `l`. This could then be removed and the value would stay centralized in our main styles, so any change would cascade when needed.

File: libs/mep/ace1209/section-metadata/section-metadata.css [hunk 6]
```diff
@@ -2,6 +2,115 @@
...
+  --bento-surface-middle: #edebeb;
+  --bento-surface-deepest: #e3e2e2;
+  --bento-shadow:
+    0 86.137px 34.455px #00000012,
+    0 35.986px 14.394px #0000000d,
+    0 19.24px 7.696px #0000000a,
+    0 10.786px 4.314px #0000000a,
+    0 5.728px 2.291px #00000008,
+    0 2.384px 0.953px #00000005;
+  --bento-shadow-none: 0 0 0 #0000;
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.css]: Any chance we can use some S2A variables?

File: libs/mep/ace1209/section-metadata/section-metadata.css [hunk 7]
```diff
@@ -287,6 +399,89 @@
...
+    }
+
+    > .explore-card .explore-card-content::after {
+      content: '';
+      position: absolute;
+      inset: 0;
+      z-index: 1;
+      border-radius: inherit;
+      pointer-events: none;
+      background: #000;
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.css]: `--s2a-color-gray-1000` maybe?

File: libs/mep/ace1209/section-metadata/section-metadata.js [hunk 8]
```diff
@@ -204,4 +204,9 @@ export default async function init(el) {
   if (metadata.layout) handleStyle(metadata.layout.text, section);
   if (metadata.images) handleImages(metadata.images?.text[0], section);
   handleStickyFocus(section);
+  if (section?.matches('.bento.stack-mobile')) {
+    import('../../../features/bento-stack.js')
+      .then(({ default: initBentoStack }) => initBentoStack(section))
+      .catch((e) => window.lana?.log(`bento-stack init failed: ${e}`, { tags: 'bento-stack', severity: 'info' }));
```
Comment (overmyheadandbody) [inline libs/mep/ace1209/section-metadata/section-metadata.js]: I'd suggest this be moved to a `handleBentoStacking` method, to follow the same pattern as above.

File: libs/features/bento-stack.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+function clearMeasurements(section) {
+  section.style.removeProperty('--card-height');
+  section.style.removeProperty('--title-height');
+  section.style.removeProperty('--gnav-offset');
+}
+
+export default function initBentoStack(section) {
+  const cards = getCards(section);
+  if (!cards.length) return;
```
Comment (overmyheadandbody) [inline libs/features/bento-stack.js]: Maybe it's a good idea to check whether `bento-stack-ready` has already been added, to avoid the possibility of re-attaching the Observer

**Review by overmyheadandbody — COMMENTED** [review]
Looking good! I have a few suggestions

File: libs/mep/ace1209/section-metadata/section-metadata.css (hunk 5 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.css]: Good catch!

File: libs/mep/ace1209/section-metadata/section-metadata.css (hunk 6 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.css]: I updated `--bento-shadow-none` to use a s2a token. The others do not have a close related value due to the transparency. The jumps are too big in the s2a tokens.

File: libs/mep/ace1209/section-metadata/section-metadata.css (hunk 3 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.css:53]: Updated.

File: libs/mep/ace1209/rich-content/rich-content.css (hunk 2 above)
Comment (rgclayton) [reply libs/mep/ace1209/rich-content/rich-content.css:554]: Updated.

File: libs/mep/ace1209/rich-content/rich-content.css (hunk 1 above)
Comment (rgclayton) [reply libs/mep/ace1209/rich-content/rich-content.css]: Updated.

File: libs/mep/ace1209/section-metadata/section-metadata.css (hunk 4 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.css]: Updated.

File: libs/mep/ace1209/section-metadata/section-metadata.css (hunk 7 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.css]: Updated.

File: libs/mep/ace1209/section-metadata/section-metadata.js (hunk 8 above)
Comment (rgclayton) [reply libs/mep/ace1209/section-metadata/section-metadata.js]: Updated.

File: libs/features/bento-stack.js (hunk 9 above)
Comment (rgclayton) [reply libs/features/bento-stack.js]: Updated.

**Comment by rgclayton** [comment]
Thanks for the feedback @overmyheadandbody!

I got all of it implemented. I also discovered an edge case race condition that I solved for. Claude suggested I add some unit tests to help with that testing so I went ahead and added those to harden the code better.

**Review by overmyheadandbody — APPROVED** [review]
Thanks for all the changes, Ryan, looking great!

**Review by SilviuLCF — APPROVED** [review]
QA verified , testing details https://jira.corp.adobe.com/browse/MWPW-199414

File: libs/mep/ace1209/rich-content/rich-content.css [hunk 10]
```diff
@@ -562,6 +543,44 @@
...
+
+    position: sticky;
+    top: var(--stack-pin-top);
+    z-index: 1;
+    margin: 0;
+    line-height: 1;
+    will-change: opacity, transform, translate;
+    animation:
+      stack-title-stagger var(--paralax-easing) both,
+      stack-title-card-match var(--paralax-easing) both;
```
Comment (zagi25) [inline libs/mep/ace1209/rich-content/rich-content.css]: Typo, I don't see an example of this on the test page, but animation won't work because of the undefined CSS variable. 
```suggestion
      stack-title-stagger var(--parallax-easing) both,
      stack-title-card-match var(--parallax-easing) both;
```

**Review by zagi25 — APPROVED** [review]
Approving, since rest of the code looks good, but we should not merge before typo is fixed.

File: libs/mep/ace1209/rich-content/rich-content.css (hunk 10 above)
Comment (rgclayton) [reply libs/mep/ace1209/rich-content/rich-content.css]: Good catch! Thank you!

Comment (rgclayton) [reply libs/mep/ace1209/rich-content/rich-content.css]: Updated.
